src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -133,13 +136,25 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mUpdateVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (update.softwareVersion > currentSoftwareVersion)
+        {
+            ChipLogDetail(SoftwareUpdate, "Updating to newer version");
+            MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
+            CopySpanToMutableSpan(update.updateToken, updateToken);
+            requestorCore->mUpdateVersion = update.softwareVersion;
+            requestorCore->mUpdateToken   = updateToken;
+
+            requestorCore->mOtaRequestorDriver->UpdateAvailable(update,
+                                                                System::Clock::Seconds32(response.delayedActionTime.ValueOr(0)));
+        }
+        else
+        {
+            ChipLogDetail(SoftwareUpdate, "Already up to date");
```

Comment:
Same here - would be more helpful for debugging to say something like "version __ is older than current __, not updating"

Reply:
older or the same, right?

Reply:
Done.